pkg/state/cluster.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"sync"
+
+	"github.com/Masterminds/semver"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kubermatic/kubeone/pkg/apis/kubeone"
+)
+
+type Cluster struct {
+	ControlPlane []Host
+	Workers      []Host
+	Lock         sync.Mutex
+}
+
+type Host struct {
+	ContainerRuntime ComponentStatus
+	Hostname         string
+	IsInCluster      bool
+	Kubeconfig       []byte
+	Kubernetes       ComponentStatus
+	OS               kubeone.OperatingSystemName
+	PrivateAddress   string
+	PublicAddress    string
+}
+
+type ComponentStatus struct {
+	Version *semver.Version
+	Status  int64
+}
+
+const (
+	SystemDStatusUnknown    = 1 << iota // systemd unit unknown
+	SystemdDStatusDead                  // systemd unit dead
+	SystemDStatusRestarting             // systemd unit restarting
+	ComponentInstalled                  // installed (package, or direct download)
```

Comment:
Why is this one in the middle of other `systemd` related consts?

Reply:
it's just another status, it's free form name.